### PR TITLE
Updates to #1465 using miniconda everywhere

### PR DESCRIPTION
**Pull request by gforsyth**
This builds on @stockninja and @santagada 's work in #1465 and #1442, using miniconda for all CI builds on travis except for "nightly"

I'm breaking it out into a separate PR to check that a) it works and b) that it's faster (which I think it should be, but let's find out).  

Also it's a bit verbose at the moment, if this works I'll switch over to using an `environment.yml` file for `conda`

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
```
Comment (scopatz) [inline install.sh]: Do we still need this install file?

**Comment by scopatz** [comment]
Thanks @gforsyth!

File: install.sh (hunk 1 above)
Comment (gforsyth) [inline install.sh]: nope, I'm going to remove it.

Comment (scopatz) [inline install.sh]: Great!

**Comment by santagada** [comment]
is it supposed to only pass on nightly?

**Comment by gforsyth** [comment]
Not by design :)

**Comment by gforsyth** [comment]
@scopatz -- seems like miniconda is a little slower than just using travis python on Linux.  We can save around 40 seconds if we don't use miniconda on the linux builds.

also, the vox tests are a problem since `venv` and conda really don't get along very well.

**Comment by scopatz** [comment]
> @scopatz -- seems like miniconda is a little slower than just using travis python on Linux. We can save around 40 seconds if we don't use miniconda on the linux builds.

I am impartial on this.  It might be nice to have faster tests, but it is also nice have consistency.  But I guess we already switch for nightlies....  Either way.

> also, the vox tests are a problem since venv and conda really don't get along very well.

Is there any way to skip these if we know we are in a conda env?  Maybe check if `__file__` or the path contains the word conda? Or that the `conda` cli is on the $PATH?

**Comment by gforsyth** [comment]
wtf... so it looks like conda's python3.5 builds don't include `ensurepip` which is part of the standard library.  that seems like a bug to me.

**Comment by scopatz** [comment]
I think this is one of the reasons why conda-forge starting building their own python.

**Comment by gforsyth** [comment]
Yeah -- I just installed that to test it out and now I'm getting a 3 year old bug where the `sys.version` regex is wrong.  This is a fun little rabbit hole.

**Comment by scopatz** [comment]
Uggh, well, I am in favor of whatever works :)

**Comment by scopatz** [comment]
Let us know when you think this is ready!

**Comment by scopatz** [comment]
Seems like there are merge conflicts now.

**Comment by gforsyth** [comment]
@scopatz @santagada -- can you take a look at this now?  All tests passing on travis and appveyor.  Total test time is less than 6 minutes for the 5 builds, but delay between each test starting makes the total walltime closer to 10 minutes.  This may be a temporary travis thing, or something else at work.

**Comment by scopatz** [comment]
I think that this is a Travis thing because of availability of machines.  Thanks a million @gforsyth!

**Pull request merged**